api/vault/vault.go

```diff
@@ -209,8 +210,28 @@ func (vault *Vault) UserSecretMetadata(secretID secretID) (*Secret, error) {
 	return metadata, err
 }
 
+func validateFilter(filter string) error {
+	filterAllowedValues := []string{"personal", "shared", "accessible", "readable", "writable", ""}
+	isValidFilter := false
+	for _, a := range filterAllowedValues {
+		if a == filter {
+			isValidFilter = true
+			break
+		}
+	}
+	if !isValidFilter {
+		return fmt.Errorf("filter field must be one of these values %q", filterAllowedValues)
+	}
+	return nil
+}
+
 // SearchSecrets search for existing secrets
-func (vault *Vault) SearchSecrets(offset, limit int, keywords, sortkey, sortdir string) ([]Secret, error) {
+func (vault *Vault) SearchSecrets(offset, limit int, keywords, sortkey, sortdir string, filter string) ([]Secret, error) {
```

Comment:
`offset, limit int, keywords, sortkey, sortdir, filter string`

Reply:
I didnt get this ?